risc0/zkvm/src/opcode.rs

```diff
@@ -88,14 +88,18 @@ impl OpCode {
         let funct7 = (insn & 0xfe000000) >> 25;
         // log::debug!("decode: 0x{word:08X}");
 
+        let decode_error_str = || {
+            format!("malformed rv32im instruction at {insn_pc:#x}: {insn:032b} ({opcode:07b}, {funct3:03b}, {rs2:05b}, {funct7:07b})")
+        };
+
         Ok(match opcode {
             0b0000011 => match funct3 {
                 0x0 => OpCode::new(insn, insn_pc, "LB", 24, 1),
                 0x1 => OpCode::new(insn, insn_pc, "LH", 25, 1),
                 0x2 => OpCode::new(insn, insn_pc, "LW", 26, 1),
                 0x4 => OpCode::new(insn, insn_pc, "LBU", 27, 1),
                 0x5 => OpCode::new(insn, insn_pc, "LHU", 28, 1),
-                _ => unreachable!(),
+                _ => unreachable!("{}", decode_error_str()),
```

Comment:
```suggestion
                _ => bail!("{}", decode_error_str()),
```

We probably should avoid panics here

Reply:
Changed to `bail!` in [d1e8ec4](https://github.com/risc0/risc0/pull/571/commits/d1e8ec4ddfc156ea99b0049ffca0ade0d89f8918)